### ISSUE TITLE
Highlight not working neovim

### ISSUE THREAD

**Issue by DarkKronicle** [postsolar/tree-sitter-kanata#1]
Hello! I'm having some difficulty getting the highlighting to work in neovim. I've installed from latest commit, and have moved the `queries/highlights.scm` to `queries/kanata/highlights.scm` in a neovim sourced folder so that treesitter will properly find it.

When I open up any kanata file, this message gets spammed several hundred times. The parsing of the tree seems to work though (using `:InspectTree`)
![image](https://github.com/postsolar/tree-sitter-kanata/assets/38167691/21e6465e-3898-44b3-ac10-5e3c70b32107)

```
Error in decoration provider treesitter/highlighter.line:
Error executing lua: ...d-0.10.0/share/nvim/runtime/lua/vim/treesitter/query.lua:331: couldn't parse regex: Vim:E866: (NFA regexp) Misplaced @
stack traceback:
	[C]: in function 'regex'
	...d-0.10.0/share/nvim/runtime/lua/vim/treesitter/query.lua:331: in function '__index'
	...d-0.10.0/share/nvim/runtime/lua/vim/treesitter/query.lua:348: in function 'handler'
	...d-0.10.0/share/nvim/runtime/lua/vim/treesitter/query.lua:767: in function 'fn'
	...pped-0.10.0/share/nvim/runtime/lua/vim/func/_memoize.lua:58: in function 'match_preds'
	...d-0.10.0/share/nvim/runtime/lua/vim/treesitter/query.lua:883: in function 'iter'
	....0/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:310: in function 'fn'
	....0/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:210: in function 'for_each_highlight_state'
	....0/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:291: in function 'on_line_impl'
	....0/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:367
```

My guess is it is [this line](https://github.com/postsolar/tree-sitter-kanata/blob/a6213d06ea6efa432702bbbc6b4c5dcddc21df2a/queries/highlights.scm#L59). Running `:echo matchstr('@hello', '\@\.\+')` produces the same error.

Looking into it a bit, it seems that vim has some weird regex syntax. Using 
```
  (#match? @string.special.symbol "\\V@\\.\\+"))
```
Seems to fix the issue

**Comment by postsolar** [comment]
Hello, and thank you for investigating this

I'm not a Vim/Neovim user myself so I didn't test the queries on it, in fact they're not meant to be compatible with Vim at all because the capture group names (`@string`, `@keyword`, etc) are different across different editors

Is it the only issue you've found with these queries? If so, we could perhaps create a separate `vim_queries/` directory in the repo to be used by Vim users, with proper capture groups names and proper regex

**Comment by postsolar** [comment]
Yes, I can reproduce it. Looks like it's not Vim-specific but a problem with block comments parsing. The contents of the comments [shouldn't be parsed at all](https://github.com/postsolar/tree-sitter-kanata/blob/a6213d06ea6efa432702bbbc6b4c5dcddc21df2a/grammar.js#L23), but for some reason they do :thinking:

**Comment by Lewenhaupt** [comment]
Yeah I get the same, @postsolar did you ever find a solution?

**Comment by postsolar** [comment]
@Lewenhaupt 

Yes, the issue is simply that it wasn't implemented yet. I elaborated on this a little bit above in the thread:

> in fact [the queries are] not meant to be compatible with Vim at all because the capture group names (`@string`, `@keyword`, etc) are different across different editors

One would need to write a queries file. If anyone's up for a PR (not a particularly involved task) I'd review it , but I won't be able to do this myself anytime soon.

**Comment by postsolar** [comment]
To aid a potential contributor, a rough plan of action would be:

1. Dig through nvim-treesitter docs and find where they describe what capture groups they use and for what
2. Attach that to a message to some AI and ask it to map [the highlights queries](https://github.com/postsolar/tree-sitter-kanata/blob/af53b4c35d6faa3b4ba7c317fb795a6eb679b42a/queries/highlights.scm) to nvim's capture groups retrieved in Step 1
3. Configure your (neo)vim to use these queries and test it
4. Put the file under `vim_queries/highlights.scm` at the project root
5. Submit the PR

I don't use vim so setting it up would take me quite a bit of time, not able to contribute it myself at the moment.

**Comment by Lewenhaupt** [comment]
Oh ok, that was quite surprising, I haven't seen any other tree-sitter grammars that have different highlight.scm files. I've looked through a bunch of them listed under nvim-treesitter (for example https://github.com/tree-sitter/tree-sitter-typescript/blob/master/queries/highlights.scm). Would that mean that the helix editor has to use it's own tree-sitter grammar for typescript?
Appears that helix might do overrides where the one for the grammar is built for nvim: https://github.com/helix-editor/helix/discussions/3020

**Comment by postsolar** [comment]
Historically at least, official Tree-sitter itself, Helix and Neovim would _each_ have their own capture groups. In the past few years, some work was going on to unify these, but I don't know if it's already fully done. At least Helix and Tree-sitter have the same groups, but I'm not sure if nvim-treesitter is there yet. Zed has (had?) its own quirks too.

My other grammar [tree-sitter-purescript](https://github.com/postsolar/tree-sitter-purescript/tree/f541f95ffd6852fbbe88636317c613285bc105af/vim_queries) does contain a separate folder for nvim-treesitter-compatible queries. On the other hand, yes, it is rare, because (non-exhaustive and speculative):
1. Grammar maintainers and nvim-treesitter integration maintainers are often different people
2. Containing nvim-treesitter-specific code in the repo is a bit weird or out of scope
3. nvim-treesitter stores its own queries in its own repo anyways

> Would that mean that the helix editor has to use it's own tree-sitter grammar for typescript?

No, not really, but yes it has to use its own _queries_ [and it does](https://github.com/helix-editor/helix/tree/8acfc55280460402e106204dd9d2d0afd2c85e93/runtime/queries). The _grammars_ used across editors are the same, but it's the syntax for query files that differs. And therefore, the query files themselves. So each integration uses its own queries — Helix, Neovim (nvim-treesitter), Zed, Kakoune, Emacs, etc.

**Comment by postsolar** [comment]
I guess it might be useful to think of queries as of configurations for the "program" (the grammar). So a grammar author has their own vision of the right "configuration", an editor distribution has their own, and the final user might even have their own too (e.g. I tweak some Helix queries, and some people tweak queries of tree-sitter-kanata). This is why the state of the ecosystem might seem somewhat messy.

**Comment by CyanideData2** [comment]
After testing for a bit I fount out that the only issue with the original highlight.scm file was some non-escaped characters (nvim didn't like line 59 `"@+"`). I've submitted a pull request with a fix in `vim_queries/highlights.scm`.
